lib/src/issue.rs

```diff
@@ -0,0 +1,138 @@
+/// repositories and are uniquely identified by an id.
+///
+pub struct Issue<'r> {
+    repo: &'r git2::Repository,
+    id: Oid,
+}
+
+impl<'r> Issue<'r> {
+    /// Create a new handle for an issue with a given id
+    ///
+    pub fn new(repo: &'r git2::Repository, id: Oid) -> Self {
+        Issue { repo: repo, id: id }
+    }
+
+    /// Get the issue's id
+    ///
+    pub fn id(&self) -> Oid {
+        self.id
+    }
+
+    /// Get possible heads of the issue
+    ///
+    /// Returns the head references from both the local repository and remotes
+    /// for this issue.
+    ///
+    pub fn heads(&self) -> Result<References<'r>> {
+        let glob = format!("**/dit/{}/head", self.unique_ref_part());
+        self.repo
+            .references_glob(&glob)
+            .chain_err(|| EK::CannotFindIssueHead(self.id))
+    }
+
+    /// Get the local issue head for the issue
+    ///
+    /// Returns the head reference of the issue from the local repository, if
+    /// present.
+    ///
+    pub fn find_local_head(&self) -> Result<Reference<'r>> {
+        let refname = format!("refs/dit/{}/head", self.unique_ref_part());
+        self.repo
+            .find_reference(&refname)
+            .chain_err(|| EK::CannotFindIssueHead(self.id))
+    }
+
+    /// Get the leaf references for the issue
+    ///
+    /// Returns the leaf references for the issue from both the local repository
+    /// and remotes.
+    ///
+    pub fn issue_leaves(&self) -> Result<References<'r>> {
+        let glob = format!("**/dit/{}/leaves/*", self.unique_ref_part());
+        self.repo
+            .references_glob(&glob)
+            .chain_err(|| EK::CannotGetReferences(glob))
+    }
+
+    /// Get all local references for the issue
+    ///
+    /// Return all references associated with the issue from the local
+    /// repository.
+    ///
+    pub fn local_refs(&self) -> Result<References<'r>> {
+        let glob = format!("refs/dit/{}/**", self.unique_ref_part());
+        self.repo
+            .references_glob(&glob)
+            .chain_err(|| EK::CannotGetReferences(glob))
+    }
+
+    /// Get a revwalk for traversing all messages of the issue
+    ///
+    /// The sorting of the revwalk will be set to "topological".
+    ///
+    pub fn message_revwalk(&self) -> Result<git2::Revwalk<'r>> {
+        let glob = format!("**/dit/{}/**", self.unique_ref_part());
+        self.repo
+            .revwalk()
+            .and_then(|mut revwalk| {
+                // The iterator will iterate over all the messages in the tree
+                // spanned but it will halt at the initial message.
+                revwalk.push_glob(glob.as_ref())?;
+                self.repo
+                    .find_commit(self.id)
+                    .and_then(|commit| commit.parent_id(0))
+                    .ok() // the initial message having no parent is not unusual
+                    .map(|parent| revwalk.hide(parent))
+                    .unwrap_or(Ok(()))?;
```

Comment:
I find that line a bit hard to follow (as in "why a default value here... we do not use it"). That would be clearer if we'd use a `let _ = self.repo` at the beginnging of this block of code... just a readability thing!